### ISSUE TITLE
Runtime error when static property of a decorated class references the class

### ISSUE THREAD

**Issue by zamb3zi** [microsoft/TypeScript#51570]
# Bug Report

### 🔎 Search Terms

static initialization decorator runtime

### 🕗 Version & Regression Information

- This changed between versions 4.5.5 and 4.6.4 when static properties were changed to be static initialization blocks

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=9#code/MYewdgzgLgBAJgU2ALhgYQDYEMIQCJIgBOWUxMAvDMBhJQHzW0DcAsAFAACiwTOdAQRgBvGNFIBLXgA9KMMAgDuMIQF8gA)

### 💻 Code

```ts
const dec: ClassDecorator = cls => cls;
@dec class A { static x = new A }
```

### 🙁 Actual behavior

No compilation error as expected, but this generates a runtime error of "undefined is not a constructor".

### 🙂 Expected behavior

Either a compilation error, or no runtime error.

**Comment by dongshimou** [comment]
same error.
```
//...
export class App {
    // static readonly inst: App = new App(); //  static inst = new _a(); compiled in js file
    static readonly inst: App = new this();
    // other code... run
}
App.inst.run();
```
ts code.
in `ES2022`.
When I write like this `static readonly inst: App = new App()`, 
the js file will compile to`static inst = new _a();`,  
error is `undefine is not a constructor`.
I manually changed js file to ` static inst = new App()` .
it's ok, everything is fine.

In the end I had to change it to this ` static readonly inst: App = new this(); `
